### PR TITLE
[HOPS-1516] stop NN and RM when the db is dead

### PR DESCRIPTION
**Pull request by berthoug**
## Make sure there is no duplicate PR for this issue

* **Please check if the PR fulfills these requirements**
- [x] Tests for the changes have been added and passed (for bug fixes / features)
- [x] HOPS JIRA issue has been opened for this PR
- [x] All commits have been squashed down to a single commit
- [x] The commit message has the following format: [HOPS-XXX] message

* **Post a link to the associated JIRA issue**
https://hopshadoop.atlassian.net/browse/HOPS-1516

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
bug fix

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:
depends on hopshadoop/hops-metadata-dal#181 and hopshadoop/hops-metadata-dal-impl-ndb#221

### CODE REVIEW

File: hadoop-hdfs-project/hadoop-hdfs/src/main/java/io/hops/transaction/context/RootINodeCache.java [hunk 1]
```diff
@@ -103,9 +104,9 @@ public Object performTask() throws IOException {
             LOG.debug("RootCache: root does not exist.");
           }
         } catch (InterruptedException e) {
-          e.printStackTrace();
+          LOG.warn(e);
```
Comment (kouzant) [inline hadoop-hdfs-project/hadoop-hdfs/src/main/java/io/hops/transaction/context/RootINodeCache.java]: Shouldn't we also interrupt the current thread if we received an InterruptedException?

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/DBRMStateStore.java [hunk 2]
```diff
@@ -826,4 +849,69 @@ public Object performTask() throws StorageException {
...
+      shouldRmRun = false;
+    }
+
+    void checkAvailableResources() {
+      int tries = 0;
+      Throwable lastThrowable = null;
+      while (tries < maxDBTries) {
+        try {
+          if (!RMStorageFactory.hasResources(databaseResourcesThreshold)) {
+            throw new IOException("not enoug resources in the database");
```
Comment (kouzant) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/DBRMStateStore.java]: Typo

File: hadoop-hdfs-project/hadoop-hdfs/src/main/java/io/hops/transaction/context/RootINodeCache.java (hunk 1 above)
Comment (berthoug) [reply hadoop-hdfs-project/hadoop-hdfs/src/main/java/io/hops/transaction/context/RootINodeCache.java]: Yea, I was a little lazy and didn't want to take the risk of changing the behavior of the class, but I will clean that up.

**Comment by berthoug** [comment]
tests: https://bbc1.sics.se/jenkins/view/pull-manual/job/hops-pull-manual/118/

**Pull request merged**